### PR TITLE
homed: report actual home dir access mode and fs type in effect

### PR DESCRIPTION
**Pull request by poettering**
So far we only reported access mode and fs type that is supposed to be used for a home dir when it is created. Let's also report what *actually* is currently used, as part of the "status" part of the user record. 

Also split out of #21253

### CODE REVIEW

File: src/home/homed-home.c [hunk 1]
```diff
@@ -2349,15 +2346,81 @@ static int home_get_disk_status_directory(
...
+                Home *h,
+                HomeState state,
+                uint64_t *ret_disk_size,
+                uint64_t *ret_disk_usage,
+                uint64_t *ret_disk_free,
+                uint64_t *ret_disk_ceiling,
+                uint64_t *ret_disk_floor) {
+
+        assert(h);
+
```
Comment (bluca) [inline src/home/homed-home.c:2409]: `assert(h->record)`

File: docs/USER_RECORD.md [hunk 2]
```diff
@@ -863,6 +863,11 @@ on removable media the delay is selected very low to minimize the chance the
 home directory remains in unclean state if the storage device is removed from
 the system by the user).
 
+`accessMode` → The access mode currently in effect for the home directory iself.
+
+`fileSystemType` → The file system type backing the home directory: a short
+string, such as "btrfs", "ext4", "xfs".
```
Comment (bluca) [inline docs/USER_RECORD.md:869]: do we have an authoritative list we can point to beside these examples? or is this the one?

**Review by bluca — APPROVED** [review]

File: docs/USER_RECORD.md (hunk 2 above)
Comment (poettering) [reply docs/USER_RECORD.md:869]: So right now homed only supports these three file systems inside the LUKS volumes, hence I added support for them, but nothing else.

I figure sooner or later we should have a table somewhere which can translate the various file system magics into their string identifiers, using the same names as /proc/filesystems would use, building on the infra we have in src/basic/filesystems.[ch] already (and which already looks at the magic but doesn't include it in the generated tables). 

Once we have that we should drop this manual translation function and just use that.

I'll add an explicit TODO list item for this. It requires some work and I don't really think it fits into this PR.

**Comment by poettering** [comment]
Force pushed a new version, with the suggested assert(), and the mentioned TODO list entry. Upgrading green label.

**Pull request merged**